test/integration/StreamEndpoints.test.js

```diff
@@ -36,6 +52,35 @@ describe('StreamEndpoints', () => {
                 assert.equal(stream.name, name)
             }))
 
+        it('createStream with private key', () => clientPrivateKey.createStream({
```

Comment:
We don't want to repeat every endpoint test with every authentication method, but then again we want to test the automatic login. How to accomplish that and where should these tests go?

Let's start from figuring out what class contains the code we're testing. In this case, I guess it's actually in `utils.js` (in `authFetch`, right? Could we write a test for that? Maybe it's a unit test even, as we can pass a mock `session` to it?

Reply:
Could we say that all integration tests which are related to authentication should be in `Session.test.js`? This way I could remove the additional clients created in this file and test some of these endpoints in `Session.test.js` with the different existing clients in that file.

Reply:
Well basically there's 2 things happening that should get tested:

1. `Session` can accept different kinds of authentication (username/password, apiKey, sessionToken). That functionality (and hence those tests cases) should be in `Session.test.js` (and IIRC they are already!).

2. `authFetch` asks for the token from `session` and uses it in the request. If the request returns 400 or 401, it forces a new token and retries. This functionality is in `utils.js` so we should have a `utils.test.js` that tests it.

If you look at the code in `StreamEndpoints`, it mainly just passes the `session` object around, so that code doesn't care about what type of credentials are contained within the session object. So I think we can ignore the different credential types in the endpoint tests (just pick one), and instead make sure functionalities 1. and 2. are covered in respective test classes.